spec/spec_helper.rb

```diff
@@ -280,3 +280,11 @@ def initialize(*args, &block)
 
 # Helper matchers
 RSpec::Matchers.define_negated_matcher :not_be, :be
+
+# The Ruby Timeout class uses a long-lived class-level thread that is never terminated.
+# Creating it early here ensures tests that tests that check for leaking threads are not
+# triggered by the creation of this thread.
+#
+# This has to be one once for the lifetime of this process, and was introduced in Ruby 3.1.
+# Before 3.1, a thread was created and destroyed on every Timeout#timeout call.
+Timeout.ensure_timeout_thread_created if Timeout.respond_to?(:ensure_timeout_thread_created)
```

Comment:
Since you mentioned this was Marco's suggestion, I wonder what changed from https://github.com/DataDog/dd-trace-rb/pull/2991 ?

I remember we had this in the past and removed it again :eyes:

Reply:
I don't believe the call to `ensure_timeout_thread_created` was removed because it was causing any problems. #2991 was attempting to solve the timeout thread being a false positive in the leaky thread report. In addition to the call to `ensure_timeout_thread_created` the old code was also [assigning a thread group to the timeout thread](https://github.com/DataDog/dd-trace-rb/pull/2991/files#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94L290-L298). This thread group was used to filter it out of the leaky thread report. However, the timeout code itself [overwrites the thread group with its own](https://github.com/ruby/timeout/blob/54835eaf4036c9f10503541d8c2bcf0b7bcdd6ab/lib/timeout.rb#L120). #2991 added [this check for the thread group](https://github.com/DataDog/dd-trace-rb/pull/2991/files#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R146-R147) set in the timeout code, so it doesn't get reported as leaky. 

I believe the call to `ensure_timeout_thread_created` was removed as it was seen as unnecessary and the assertion of threads being present before/after certain conditions in tests wasn't taken into consideration at the time. @marcotc feel free to correct me if I mischaracterized your thinking here.